dll/kernel32.cpp

```diff
@@ -10,15 +10,42 @@
 #include <sys/stat.h>
 
 namespace kernel32 {
+	char *WideStringToString(const uint16_t *src) {
+		char *res = NULL;
+
+		int len = 0;
+		if (src != NULL) {
+			while (src[len] != 0) {
+				len++;
+			};
+		}
+		res = (char *)malloc(len + 1);
```

Comment:
returning an std::string as you had it before makes more sense to me. It should be just as efficient and clarifies ownership